src/radio.rs

```diff
@@ -161,23 +159,62 @@ impl Phy {
             state: State::Idle,
         };
 
-        crate::crit_sect::locked(|cs| {
-            let prev_isr_data = ISR_DATA.borrow(cs).replace(Some(isr_data));
-            assert!(prev_isr_data.is_none());
-        });
+        let isr_data = Mutex::new(RefCell::new(isr_data));
 
-        Self {
-            isr_data: &ISR_DATA,
-        }
+        Self { isr_data }
+    }
+
+    fn get_global_instance_raw() -> &'static OnceCell<Phy> {
+        static PHY: OnceCell<Phy> = OnceCell::new();
+        &PHY
+    }
+
+    /// Set this Phy as global instance.
+    ///
+    /// Panics if global instance was already set.
+    pub fn set_as_global_instance(self) -> &'static Self {
+        Self::get_global_instance_raw()
+            .try_insert(self)
+            .expect("Global Phy already initialized")
+    }
+
+    /// Create new global instance.
+    ///
+    /// Panics if global instance was already set.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #[macro_use] extern crate nrf_radio;
+    /// # missing_test_fns!();
+    /// # fn main() {
+    ///   use nrf52840_hal::pac::Peripherals;
+    ///   use nrf_radio::radio::Phy;
+    ///
+    ///   let peripherals = Peripherals::take().unwrap();
+    ///
+    ///   let phy = Phy::new_global_instance(&peripherals.RADIO);
+    /// # }
+    /// ```
+    pub fn new_global_instance(radio: &RadioRegisterBlock) -> &'static Self {
+        Self::new(radio).set_as_global_instance()
+    }
+
+    /// Get global instance.
+    ///
+    /// Panics is global instance is not set.
+    pub fn get_global_instance() -> &'static Self {
+        Self::get_global_instance_raw()
+            .get()
+            .expect("Global Phy not initialized")
     }
```

Comment:
A reference to `Phy` should be passed between modules to ensure only one can access `Phy` at the time (according to scheduling radio operations). So I don't think we should have a function that any module can use to get reference to the global `Phy` instance.
```suggestion
```

Reply:
Ok, so the idea is that in the future, `Phy` will have items not used from ISR? Also, how ISR_DATA will be related to Phy? Assuming multiple ISR_DATA instances are possible, will there be multiple `Phy` instances as well?

Reply:
The idea for `Phy` implementation is like this: we should assume multiple radios in a single system, so multiple `Phy` instances possible, each associated with a separated ISR, so with separated `ISR_DATA`.

I cannot rule out the possibility of some fields in `Phy` structure which do not have to be accessible from ISRs, so to achieve correct encapsulation, such fields shall be inaccessible from ISRs (out of `ISR_DATA`).

Moreover, there might be multiple users of one of the `Phy`s (or both) with a time-slicing algorithm passing `Phy` access from one user to another. I'm thinking of a higher-layer module like `PhyAccess` containing `&'static Phy` and the users would pass ownership of `PhyAccess` from one to another according to the time-slicing. This way we would ensure that there is no concurrent access to the peripheral.